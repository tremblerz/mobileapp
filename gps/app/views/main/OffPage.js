import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  Dimensions,
  ImageBackground,
  StatusBar,
  Text,
  View,
} from 'react-native';
import { SvgXml } from 'react-native-svg';

import backgroundImage from '../../../../shared/assets/images/launchScreenBackground.png';
import StateUnknown from '../../../../shared/assets/svgs/stateUnknown';
import { Button } from '../../components/Button';
import { Typography } from '../../components/Typography';
import { Theme } from '../../constants/themes';
import { MayoButton } from './MayoButton';
import { styles } from './style';

export const OffPage = () => {
  const { t } = useTranslation();
  const navigation = useNavigation();
  const buttonLabel = t('label.home_enable_location');
  const size = Dimensions.get('window').height;
  return (
    <Theme use='violet'>
      <ImageBackground source={backgroundImage} style={styles.backgroundImage}>
        <StatusBar
          barStyle='light-content'
          backgroundColor='transparent'
          translucent
        />
        <View style={styles.pulseContainer}>
          <SvgXml
            xml={StateUnknown}
            width={size ? size : 80}
            height={size ? size : 80}
          />
        </View>

        <View style={styles.mainContainer}>
          <View style={styles.contentAbovePulse}>
            <Typography style={styles.mainTextAbove} />
            <Typography style={styles.subsubheaderText} />
          </View>
          <View style={styles.contentBelowPulse}>
            <Text style={styles.mainTextBelow}>
              {t('label.home_setting_off_header')}
            </Text>
            <Typography style={styles.subheaderText}>
              {t('label.home_setting_off_subtext')}
            </Typography>
            <Button
              label={buttonLabel}
              onPress={() => navigation.navigate('SettingsScreen', {})}
              style={styles.buttonContainer}
            />
          </View>
        </View>
        <MayoButton />
      </ImageBackground>
    </Theme>
  );
};
