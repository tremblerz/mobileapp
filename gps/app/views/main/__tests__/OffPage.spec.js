import { render } from '@testing-library/react-native';
import React from 'react';

import { OffPage } from '../OffPage';

it('setting off matches snapshot', () => {
  const { asJSON } = render(<OffPage />);

  expect(asJSON()).toMatchSnapshot();
});
