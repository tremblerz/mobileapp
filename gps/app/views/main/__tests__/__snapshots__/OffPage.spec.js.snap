// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setting off matches snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../shared/assets/images/launchScreenBackground.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 0,
          "top": "-13%",
        }
      }
    >
      <SvgXml
        height={1334}
        width={1334}
        xml="<svg width=\\"850\\" height=\\"850\\" viewBox=\\"0 0 850 850\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"85.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"149.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"217.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"283.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"352.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"424.5\\" stroke=\\"white\\"/>
<path d=\\"M425 465C447.091 465 465 447.091 465 425C465 402.909 447.091 385 425 385C402.909 385 385 402.909 385 425C385 447.091 402.909 465 425 465Z\\" fill=\\"white\\"/>
<path d=\\"M425 457C442.673 457 457 442.673 457 425C457 407.327 442.673 393 425 393C407.327 393 393 407.327 393 425C393 442.673 407.327 457 425 457Z\\" fill=\\"#FF5656\\"/>
<path d=\\"M426.373 429.756V427.292C429.005 426.816 431.581 424.968 431.581 421.608C431.581 418.304 429.145 416.12 425.421 416.12C421.725 416.12 419.765 418.276 419.065 420.628L421.725 421.692C422.173 420.012 423.377 418.78 425.309 418.78C427.129 418.78 428.333 419.676 428.333 421.524V421.916C428.333 424.268 426.457 425.304 423.433 425.416V429.756H426.373ZM424.973 436.28C426.681 436.28 427.325 435.468 427.325 434.404V433.732C427.325 432.668 426.681 431.856 424.973 431.856C423.265 431.856 422.621 432.668 422.621 433.732V434.404C422.621 435.468 423.265 436.28 424.973 436.28Z\\" fill=\\"white\\"/>
</svg>"
      />
    </View>
    <View
      style={
        Object {
          "height": "100%",
          "left": 0,
          "paddingBottom": 12,
          "paddingHorizontal": "12%",
          "position": "absolute",
          "right": 0,
          "top": "-10%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "paddingBottom": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Medium",
              "fontSize": 28,
              "fontWeight": "normal",
              "lineHeight": 34,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Light",
              "fontSize": 16,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingTop": 80,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Medium",
              "fontSize": 26,
              "lineHeight": 34,
              "marginBottom": 24,
              "textAlign": "center",
            }
          }
        >
          Unknown
        </Text>
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          We can’t tell if you’re at risk unless you enable location history in the settings screen.
        </Text>
        <View
          accessibilityLabel="Enable Location Data"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#FFF",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderWidth": 2,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "center",
              "marginTop": 24,
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#4051DB",
                "fontFamily": "IBMPlexSans-Medium",
                "fontSize": 20,
                "fontWeight": "normal",
                "lineHeight": 40,
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            Enable Location Data
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "flex-end",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            testID="MayoLinkHeading"
            use="body1"
          >
            More COVID-19 information
          </Text>
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            testID="MayoLinkLabel"
            use="body1"
          >
            from the Mayo Clinic
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../shared/assets/images/foreArrow.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
